website_and_docs/content/documentation/grid/advanced_features/customize_node.en.md

```diff
@@ -0,0 +1,206 @@
+2. Add the below two dependencies to your sample project.
+  * [org.seleniumhq.selenium/selenium-java](https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-java)
+  * [org.seleniumhq.selenium/selenium-grid](https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-grid/4.6.0)
+3. Add your customized node to the project.
+4. Build an [uber jar](https://imagej.net/develop/uber-jars) if you would like to just use `java -jar` to start the node.
+5. Now start the node using the command:
+
+```bash
+java -jar selenium-server-4.6.0.jar node \
+--node-implementation org.seleniumhq.samples.CommentatingNode
+```
+
+Here's a sample that just prints some messages on to the console whenever there's an activity of interest (session created, session deleted, a webdriver command executed etc.,) on the node.
+
+```java
+package org.seleniumhq.samples;
+
+import java.net.URI;
+import java.util.UUID;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.NoSuchSessionException;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.grid.config.Config;
+import org.openqa.selenium.grid.data.CreateSessionRequest;
+import org.openqa.selenium.grid.data.CreateSessionResponse;
+import org.openqa.selenium.grid.data.NodeId;
+import org.openqa.selenium.grid.data.NodeStatus;
+import org.openqa.selenium.grid.data.Session;
+import org.openqa.selenium.grid.log.LoggingOptions;
+import org.openqa.selenium.grid.node.HealthCheck;
+import org.openqa.selenium.grid.node.Node;
+import org.openqa.selenium.grid.node.local.LocalNodeFactory;
+import org.openqa.selenium.grid.security.Secret;
+import org.openqa.selenium.grid.security.SecretOptions;
+import org.openqa.selenium.grid.server.BaseServerOptions;
+import org.openqa.selenium.internal.Either;
+import org.openqa.selenium.remote.SessionId;
+import org.openqa.selenium.remote.http.HttpRequest;
+import org.openqa.selenium.remote.http.HttpResponse;
+import org.openqa.selenium.remote.tracing.Tracer;
+
+public class DecoratedLoggingNode extends Node {
+
+  private Node node;
+
+  protected DecoratedLoggingNode(Tracer tracer, URI uri, Secret registrationSecret) {
+    super(tracer, new NodeId(UUID.randomUUID()), uri, registrationSecret);
+  }
+
+  public static Node create(Config config) {
+    LoggingOptions loggingOptions = new LoggingOptions(config);
+    BaseServerOptions serverOptions = new BaseServerOptions(config);
+    URI uri = serverOptions.getExternalUri();
+    SecretOptions secretOptions = new SecretOptions(config);
+    Node node = LocalNodeFactory.create(config);
+    DecoratedLoggingNode wrapper = new DecoratedLoggingNode(loggingOptions.getTracer(),
+        uri, secretOptions.getRegistrationSecret());
+    wrapper.node = node;
+    return wrapper;
+  }
+
+  @Override
+  public Either<WebDriverException, CreateSessionResponse> newSession(
+      CreateSessionRequest sessionRequest) {
+    System.err.println("Before newSession()");
```

Comment:
Any particular reason why `.err.` is used instead of `.out.`?

Reply:
No specific reason @diemol Just old habits that's all

Reply:
I'd prefer to leave it with `out`, please.